### PR TITLE
SAK-49722: Problem alphabetical order of the student names in the "Section info" tool - Correction for correct order

### PR DESCRIPTION
**Pull request by victorGomollon**
[SAK-49722: Problem alphabetical order of the student names in the "Section info" tool ](https://sakaiproject.atlassian.net/browse/SAK-49722)

### CODE REVIEW

**Review by jesusmmp — APPROVED** [review]

File: sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java [hunk 1]
```diff
@@ -74,7 +76,15 @@ public class EditManagersBean extends CourseDependentBean implements Serializabl
 		public int compare(Object o1, Object o2) {
 			ParticipationRecord manager1 = (ParticipationRecord)o1;
 			ParticipationRecord manager2 = (ParticipationRecord)o2;
-			Collator collator = Collator.getInstance();
+			Collator collator;
+			try {
+				collator = new RuleBasedCollator(((RuleBasedCollator) Collator.getInstance()).getRules().replaceAll("<'\u005f'", "<' '<'\u005f'"));
```
Comment (ottenhoff) [inline sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: This is super obtuse. I have no idea what it does from looking at it.

Also this is done in one app while there are dozens of places in Sakai that do sorting.

Plus there are no tests!

Comment (jesusmmp) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: In other places on Sakai it's done on that way.
For unit tests please file a new jira but for the moment we are not going to implement it. Thanks!

Comment (ottenhoff) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: so what does this code do?

Comment (jesusmmp) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: This suggests modifying the sorting behavior to consider a space before an underscore in string comparisons.

Other places:

- https://github.com/sakaiproject/sakai/blame/master/assignment/api/src/java/org/sakaiproject/assignment/api/sort/UserIdComparator.java#L41
- https://github.com/sakaiproject/sakai/blame/master/assignment/api/src/java/org/sakaiproject/assignment/api/sort/AssignmentSubmissionComparator.java#L58

Comment (ottenhoff) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: why not make one user comparator (with tests) and be done with this madness?

Comment (jesusmmp) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: Feel free to make a new PR and make changes as you see fit.

Comment (ottenhoff) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: The goal is for quality, well-tested code that is understandable by all. Quick fixes that don't use centralized methods and don't have tests are not what we are looking for as a community.

Comment (ottenhoff) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: I don't want to revert it.

Comment (jesusmmp) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: So what do we do then?

Comment (ottenhoff) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: Nothing. I just want us all to be on the same page that the goal is to solve things holistically, ideally with tests.

Comment (jesusmmp) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: We have provided many improvements in the past with unit tests. You can ask @ern.
Nowadays, we don't have enough resources (people) to do these unit tests.

Comment (ottenhoff) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditManagersBean.java:81]: The idea of centralized methods and tests is that it saves hours from testing code in a browser and testing in PROD. A central method with a test also ensures that no one can ever break your test with a future change.

Fixing this same issue 10 times sounds like a terrible waste of time.

**Pull request merged**